### PR TITLE
Feat/mongo db plugin poc

### PR DESCRIPTION
**Pull request by morhag90**

### CODE REVIEW

File: plugins/db-mongo/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# @amplication/plugin-db-postgres
```
Comment (abrl91) [inline plugins/db-mongo/README.md:1]: update the readme 🙂

File: plugins/db-mongo/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  beforeCreateServerDotEnv(
+    context: DsgContext,
+    eventParams: CreateServerDotEnvParams
+  ) {
+    eventParams.envVariables = [...eventParams.envVariables, ...envVariables];
+
+    return eventParams;
+  }
+
+  beforeCreateServerDockerCompose(
```
Comment (abrl91) [inline plugins/db-mongo/src/index.ts:45]: On MySQL plugin we had to change some variables (db-user, db-password, db-port).
Did you run the app with the current environment variables as-is? (the default Postgres values)
```
npm run docker:db
npm run prisma generate
npm run db:init
```

In general, I think that each db-plugin should override these variables to the expected values, for example:
Postgres: 5432 (the default)
MySQL: 3306 (the plugin overrides the default)
MongoDB: 27017 (up to you)
@yuval-hazaz, what do you think?

**Review by abrl91 — CHANGES_REQUESTED** [review]

**Review by yuval-hazaz — APPROVED** [review]

File: plugins/db-mongo/src/index.ts (hunk 2 above)
Comment (yuval-hazaz) [reply plugins/db-mongo/src/index.ts:54]: We expose these parameters in the UI so the user may override the values for their needs. the internal port on the docker is correct (27017)
I am OK with proceeding the way it is, and we will review it soon as part of the bigger pictures on how we install DB plugins from the UI and wether we keep the DB settings page and more specifically the port parameter

**Comment by yuval-hazaz** [comment]
@morhag90 if you are ready, let's proceed with publishing the plugin and test it live

File: plugins/db-mongo/src/constants.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+          environment: {
+            DB_URL: "mongodb://${DB_USER}:${DB_PASSWORD}@db:27017",
+          },
+        },
+        db: {
+          image: "mongo",
+          ports: ["${DB_PORT}:27017"],
+          environment: {
+            POSTGRES_USER: "${DB_USER}",
+            POSTGRES_PASSWORD: "${DB_PASSWORD}",
```
Comment (abrl91) [inline plugins/db-mongo/src/constants.ts:34]:
```suggestion
            MONGO_USER: "${DB_USER}",
            MONGO_PASSWORD: "${DB_PASSWORD}",
```

File: plugins/db-mongo/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
+# @amplication/plugin-db-mongo
+
+[![NPM Downloads](https://img.shields.io/npm/dt/@amplication/plugin-db-mongo)](https://www.npmjs.com/package/@amplication/plugin-db-mongo) 
+
+Use a Mongo DB database in the service generated by Amplication.
+
+## Purpose
+
+This plugin adds the required code to use Mongo database in the service generated by Amplication.
```
Comment (abrl91) [inline plugins/db-mongo/README.md:9]: add a bullet for the `package.json` and a bullet for the prisma-fields

File: plugins/db-mongo/static/package-json/package.template.json [hunk 5]
```diff
@@ -0,0 +1,75 @@
+{
```
Comment (abrl91) [inline plugins/db-mongo/static/package-json/package.template.json:1]: it should be `package.json` and not `package.template.json`

File: plugins/db-mongo/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+  beforeCreateServerDockerComposeDB(
+    context: DsgContext,
+    eventParams: CreateServerDockerComposeDBParams
+  ) {
+    context.utils.skipDefaultBehavior = true;
+    return eventParams;
+  }
+
+  async afterCreateServerDockerComposeDB(context: DsgContext) {
+    const staticPath = resolve(__dirname, "../static");
```
Comment (abrl91) [inline plugins/db-mongo/src/index.ts:94]: now, you need to put the `docker-compose.db.yml` also under a folder and change this line to:
```suggestion
    const staticPath = resolve(__dirname, "../static/docker-compose");
```

**Review by abrl91 — CHANGES_REQUESTED** [review]

File: plugins/db-mongo/static/package-json/package.template.json (hunk 5 above)
Comment (morhag90) [reply plugins/db-mongo/static/package-json/package.template.json:1]: @abrl91 ok, now i see this generate the file with that name, i will change it :)

**Review by abrl91 — APPROVED** [review]

**Pull request merged**